core/data/src/test/java/uk/gov/gchq/gaffer/data/util/ElementUtil.java

```diff
@@ -43,4 +43,22 @@ public static void assertElementEquals(final Iterable<? extends ElementId> expec
             assertThat(resultCache).containsExactlyInAnyOrderElementsOf(expectedCache);
         }
     }
+
+    public static void assertElementEqualsIncludingMatchedVertex(final Iterable<? extends ElementId> expected, final Iterable<? extends ElementId> result) {
+        assertElementEqualsIncludingMatchedVertex(expected, result, false);
+    }
+
+    public static void assertElementEqualsIncludingMatchedVertex(final Iterable<? extends ElementId> expected, final Iterable<? extends ElementId> result, final boolean ignoreDuplicates) {
+        final List<ElementId> expectedCache = Lists.newArrayList(expected);
+        final List<ElementId> resultCache = Lists.newArrayList(result);
+        if (ignoreDuplicates) {
+            assertThat(resultCache)
+                .usingRecursiveFieldByFieldElementComparator()
+                .hasSameElementsAs(expectedCache);
+        } else {
+            assertThat(resultCache)
+                .usingRecursiveFieldByFieldElementComparator()
+                .containsExactlyInAnyOrderElementsOf(expectedCache);
```

Comment:
The same applies to the code further up as well.
```suggestion
        if (ignoreDuplicates) {
            assertThat(result)
                .asList()
                .usingRecursiveFieldByFieldElementComparator()
                .hasSameElementsAs(expected);
        } else {
            assertThat(result)
                .asList()
                .usingRecursiveFieldByFieldElementComparator()
                .containsExactlyInAnyOrderElementsOf(expected);
```

Reply:
I tried this but doesn't work in [MapStore tests](https://github.com/gchq/Gaffer/actions/runs/7248721759/job/19745194192?pr=3131)

Reply:
It seems use of the Guava `Lists` library is only to perform a cast without showing any warning. Using `asList` doesn't work because it just casts to a `List` which will fail if the underlying `Iterable` wasn't one already.

The assertion can be used directly by simply casting to remove the `?` wildcard `assertThat((Iterable<ElementId>) result)`. This doesn't fail for those Map Store tests.